File: service/src/main/java/com/iberianmotorsports/service/model/Sanction.java
package com.iberianmotorsports.service.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.io.Serializable;

@Entity
@Data
@Table(name = "SANCTION")
public class Sanction implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @NotNull
    @Column(name = "lap")
    private Integer lap;

    @NotNull
    @Column(name = "penalty")
    private String penalty;

    @Column(name = "reason")
    private String reason;

    @Column(name = "in_game")
    private Boolean inGame;

    @Column(name = "sanction_points")
    private Long sanctionPoints;

    @ManyToOne
    @JoinColumns({
            @JoinColumn(name = "race_id", referencedColumnName = "race_id"),
            @JoinColumn(name = "grid_id", referencedColumnName = "grid_id")
    })
    private GridRace gridRace;
}

File: frontend/src/app/model/Sanction.ts
import {Grid} from "./Grid";


export class Sanction {
    id: number;
    lap: number;
    penalty: number;
    reason: string;
    raceId: number;
    gridId: number;
    grid: Grid;
    inGame: boolean;
    licensePoints: number;

    constructor() {
        this.id = null;
        this.lap = null;
        this.penalty = null;
        this.reason = "";
        this.gridId = null;
        this.raceId = null;
        this.grid = null;
        this.inGame = null;
        this.licensePoints = null;
    }
}
